pyvista/core/utilities/fileio.py

```diff
@@ -1056,26 +1056,35 @@ def to_meshio(mesh: DataSet) -> meshio.Mesh:
     connectivity = mesh.cell_connectivity
 
     # Generate polyhedral cell faces if any
-    polyhedral_cells = pyvista.convert_array(mesh.GetFaces())
+    if pyvista.vtk_version_info < (9, 4):
+        polyhedral_cells = pyvista.convert_array(mesh.GetFaces())
 
-    if polyhedral_cells is not None:
-        locations = pyvista.convert_array(mesh.GetFaceLocations())
-        polyhedral_cell_faces = []
+        if polyhedral_cells is not None:
+            locations = pyvista.convert_array(mesh.GetFaceLocations())
+            polyhedral_cell_faces = []
 
-        for location in locations:
-            if location == -1:
-                continue
+            for location in locations:
+                if location == -1:
+                    polyhedral_cell_faces.append([])
+                    continue
+
+                n_faces = polyhedral_cells[location]
+                i = location + 1
+                faces: list[MatrixLike[int]] = []
 
-            n_faces = polyhedral_cells[location]
-            i = location + 1
-            faces: list[MatrixLike[int]] = []
+                while len(faces) < n_faces:
+                    n_vertices = polyhedral_cells[i]
+                    faces.append(polyhedral_cells[i + 1 : i + 1 + n_vertices])
+                    i += n_vertices + 1
 
-            while len(faces) < n_faces:
-                n_vertices = polyhedral_cells[i]
-                faces.append(polyhedral_cells[i + 1 : i + 1 + n_vertices])
-                i += n_vertices + 1
+                polyhedral_cell_faces.append(faces)
+
+    else:
+        from ..cell import _get_irregular_cells
```

Comment:
Is there a public version for this? If not, I think it should or be handled by ``pyvista.convert_array()``.